app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesRepositoryFirestore.kt

```diff
@@ -0,0 +1,202 @@
+      profile: Profile,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    performFirestoreOperation(
+        db.collection(collectionPath).document(profile.uid).set(profile), onSuccess, onFailure)
+  }
+
+  /**
+   * Retrieves a profile from Firestore by its UID.
+   *
+   * @param uid The UID of the profile to retrieve.
+   * @param onSuccess A callback invoked with the profile if the operation is successful, or null if
+   *   no profile is found.
+   * @param onFailure A callback invoked with an exception if the operation fails.
+   */
+  override fun getProfileByUid(
+      uid: String,
+      onSuccess: (Profile?) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    Log.d("ProfilesRepositoryFirestore", "getProfileByUid")
+    db.collection("profiles").document(uid).get().addOnCompleteListener { task ->
+      if (task.isSuccessful) {
+        val profile = task.result?.let { document -> documentToProfile(document) }
+        onSuccess(profile)
+      } else {
+        task.exception?.let { e ->
+          Log.e("ProfilesRepositoryFirestore", "Error getting profile", e)
+          onFailure(e)
+        }
+      }
+    }
+  }
+
+  /**
+   * Deletes a profile from Firestore by its UID.
+   *
+   * @param uid The UID of the profile to delete.
+   * @param onSuccess A callback invoked if the operation is successful.
+   * @param onFailure A callback invoked with an exception if the operation fails.
+   */
+  override fun deleteProfileByUid(
+      uid: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    performFirestoreOperation(
+        db.collection(collectionPath).document(uid).delete(), onSuccess, onFailure)
+  }
+
+  /**
+   * Performs a Firestore operation and calls the appropriate callback based on the result.
+   *
+   * @param task The Firestore task to perform.
+   * @param onSuccess The callback to call if the operation is successful.
+   * @param onFailure The callback to call if the operation fails.
+   */
+  private fun performFirestoreOperation(
+      task: Task<Void>,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    task.addOnCompleteListener { result ->
+      if (result.isSuccessful) {
+        onSuccess()
+      } else {
+        result.exception?.let { e ->
+          Log.e("TodosRepositoryFirestore", "Error performing Firestore operation", e)
+          onFailure(e)
+        }
+      }
+    }
+  }
+
+  /**
+   * Converts a Firestore document to a Profile object.
+   *
+   * @param document The Firestore document to convert.
+   * @return The Profile object, or null if conversion fails.
+   */
+  private fun documentToProfile(document: DocumentSnapshot): Profile? {
+    return try {
+      val uid = document.id
+      val name = document.getString("name") ?: return null
+      val genderString = document.getString("gender") ?: return null
+      val gender = Gender.valueOf(genderString) // Convert string to Gender enum
```

Comment:
what if for x reason, the string doesn't match Gender enum ? You removed the fallback here but I think it would make sense to maintain it